gpg/sign.go

```diff
@@ -35,7 +35,17 @@ func Signer(d *dagger.Client, pubkey, privkey, passphrase string) *dagger.Contai
 		WithMountedSecret("/root/.rpmdb/privkeys/grafana.key", gpgPrivateKeySecret).
 		WithMountedSecret("/root/.rpmdb/pubkeys/grafana.key", gpgPublicKeySecret).
 		WithMountedSecret("/root/.rpmdb/passkeys/grafana.key", gpgPassphraseSecret).
-		WithExec([]string{"rpm", "--import", "/root/.rpmdb/pubkeys/grafana.key"}).
+		WithExec([]string{"/bin/sh", "-c", `
+			if grep -q "PUBLIC KEY" "/root/.rpmdb/pubkeys/grafana.key"; then
+				cp "/root/.rpmdb/pubkeys/grafana.key" "/tmp/grafana.key";
+			else
+				gpg --enarmor "/root/.rpmdb/pubkeys/grafana.key" > "/tmp/grafana.key";
+			fi;
```

Comment:
Just in case it woudn't be armored?

Reply:
Can we assume that the presence of "PUBLIC KEY" always indicates a correctly formatted ASCII-armored key?